picard/track.py

```diff
@@ -170,7 +170,12 @@ def add_file(self, file):
     def update_file_metadata(self, file):
         if file not in self.linked_files:
             return
-        file.copy_metadata(self.metadata)
+        # Run the scripts for the file to allow usage of
+        # file specific metadata and variables
+        metadata = Metadata(file.metadata)
+        metadata.update(self.orig_metadata)
+        self.run_scripts(metadata)
+        file.copy_metadata(metadata)
```

Comment:
In my local version of this patch I've been using the following sequence:
```python
        if file not in self.linked_files:
            return
        file.copy_metadata(self.orig_metadata)
        file.metadata['~extension'] = file.orig_metadata['~extension']
        self.run_scripts(file.metadata)
        file.update(signal=False)
        self.update()

```
which avoids making an extra metadata object. I think this might also work better in case of a file getting unmatched and then rematched on a different track?

Reply:
I'm not 100% sure that my version of the code is actually doing the right thing here, It's been a long time since I've gone through how the loading sequence works.

Reply:
I had some reason to run `run_scripts` before `file.copy_metadata`, but I don't remember right now. For rematching it would not make a difference, same end result.

Reply:
Now I remember: Your code breaks the preserve tags functionality under some circumstances. If the script changes a tag this will be written to the file even if you have this tag on your preserve list. The preserving is handled by file.copy_metadata. Hence the additional work to prepare the scripted metadata first before applying it.

Reply:
This was intentional - I wanted to be able to use scripts to modify the values in "preserved" tags.